Sources/Valet/Internal/SecItem.swift

```diff
@@ -228,12 +178,20 @@ internal final class SecItem {
         }
         
         if status == errSecSuccess {
-            return .success
+            // We're done!
             
         } else {
-            ErrorHandler.assert(status != errSecMissingEntitlement, "A 'Missing Entitlements' error occurred. This is likely due to an Apple Keychain bug. As a workaround try running on a device that is not attached to a debugger.\n\nMore information: https://forums.developer.apple.com/thread/4743")
-            
-            return .error(status)
+            switch KeychainError(status: status) {
+            case .couldNotAccessKeychain:
+                throw KeychainError.couldNotAccessKeychain
+
+            case .missingEntitlement:
+                throw KeychainError.missingEntitlement
+
+            default:
+                // We succeeded as long as we can confirm that the item is not in the keychain.
```

Comment:
Would all of the other errors mean that this succeeded? Seems like `errSecSuccess` is the only one that means it actually got deleted.

Reply:
This preserves existing behavior that was previously in [Keychain.swift](https://github.com/square/Valet/pull/198/files/b1d205b1c794bae016547e44372f0b1432764d32#diff-15d291e7a67cc811c64966484db0943aL166). Let's consider improving this code in a separate pass to reduce churn in this PR.

Reply:
Let's add a comment here. This does indeed look weird when we know `KeychainError` has more errors that it could handle. Also, should we make this explicit by handling all cases instead of using `default`?

Reply:
I like making it explicit. I think that makes it more clear, and removes the need for a comment. Good thinking!

Addressed in 4e9eb8b